sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -313,6 +313,42 @@ trait CheckAnalysis extends PredicateHelper {
               failAnalysis(s"Invalid partitioning: ${badReferences.mkString(", ")}")
             }
 
+          case v @ View(desc, output, child) if child.resolved && !v.sameOutput(child) =>
```

Comment:
Basically, I moved analysis exception check from `AliasViewChild`. In `EliminateView` rule, it doesn't need to check this.